### ISSUE TITLE
Provide introspection for VideoFrame types

### ISSUE THREAD

**Issue by sandersdan** [WICG/web-codecs#67]
An app should be able to determine the pixel formats that can be used to create a VideoFrame. This could be a static VideoFrame.supportedPixelFormats array, or perhaps a VideoFrame.isSupportedShape() API that allows checking the against any size limit also.

**Comment by chcunningham** [comment]
Editors note: marking 'extension', as this proposes a new non-breaking API.

**Comment by padenot** [comment]
For now, it's not particularly important because it's YUV420 or nothing (per spec), but as soon as this is extended this becomes fairly important.

**Comment by chcunningham** [comment]
In recent discussion (https://github.com/w3c/webcodecs/issues/407#issuecomment-974563456) w/ @sandersdan this seemed less important, but we're flexible.

> Aside: for VideoFrame speicifically, the current (implied) requirement is that all implementers must support construction of VideoFrame's with any VideoPixelFormat. As we add new formats, there will be implementations that are behind the spec, but that is easily discoverable (failed construction). For now we don't see a need to add a VideoFrame.isFormatSupported() type API, but we can fallback to that if needed.

**Comment by aboba** [comment]
@Djuffin @padenot This issue was opened 3.5 years ago and has not had activity in 2.5 years.  Is it obsolete?

**Comment by Djuffin** [comment]
Yes, let's close on the basis of no activity and  https://github.com/w3c/webcodecs/issues/67#issuecomment-981976402
